# HG changeset patch
# User sbr <sebastien@brissaud.name>
# Date 1379408245 -7200
#      Tue Sep 17 10:57:25 2013 +0200
# Branch fs5742_00
# Node ID fcd479561a6e2dbcfc788ad6d253681e26fd2a0a
# Parent  420d26312b6dcf662c8b30927eaf6e431d65e905
-Feature: Increase the object class limit from 32 to 255.

Before NewObject was implemented [1], NewGRFs provided eye-candy tiles as NewStation objects.
Now it make more sense to defines these tiles as NewObjects.

The limit of allowed NewObject classes [2] is easily reached as seen in FS#5742 with only 7 NewGRFs.
As had been done for the limit of NewStation classes in r24031 [3], it make sense to also increase this limit for NewObject classes.

This patch raise the limit of allowed NewObject classes from 32 to 255.

[1]: https://secure.openttd.org/vcs/hg/openttd/trunk.hg/rev/7235088d0c61
[2]: https://secure.openttd.org/vcs/hg/openttd/trunk.hg/file/dd5395efe404/src/newgrf_object.h#l49
[3]: https://secure.openttd.org/vcs/hg/openttd/trunk.hg/rev/c6ebcc917cba

diff --git a/src/newgrf_object.h b/src/newgrf_object.h
--- a/src/newgrf_object.h
+++ b/src/newgrf_object.h
@@ -43,13 +43,13 @@ DECLARE_ENUM_AS_BIT_SET(ObjectFlags)
 
 void ResetObjects();
 
 /** Class IDs for objects. */
 enum ObjectClassID {
 	OBJECT_CLASS_BEGIN   =    0, ///< The lowest valid value
-	OBJECT_CLASS_MAX     =   32, ///< Maximum number of classes.
+	OBJECT_CLASS_MAX     = 0xFF, ///< Maximum number of classes.
 	INVALID_OBJECT_CLASS = 0xFF, ///< Class for the less fortunate.
 };
 /** Allow incrementing of ObjectClassID variables */
 DECLARE_POSTFIX_INCREMENT(ObjectClassID)
 
 /** An object that isn't use for transport, industries or houses. */
